### ISSUE TITLE
[v3.7.0] (Regression) MotionStreak with RigidBody2D does not move correctly on Android

### ISSUE THREAD

**Issue by mrtzd** [cocos/cocos-engine#14300]
### Cocos Creator version

3.7.0, 3.6.3

### System information

Windows 10 Editor

### Issue description

I have a node with a MotionStreak and RigidBody2D components. When I try to move the node it does not render properly on Android devices (seems like it has wrong position and it goes offscreen).
This is the code for the demo:
```
export class Movable extends Component {
    private time: number = 0;
    private direction: number = 1;
    private pos: Vec3 = new Vec3();

    update(deltaTime: number) {
        this.time += deltaTime;
        if (this.time > 3) {
            this.time = 0;
            this.direction *= -1;
        }
        this.node.getPosition(this.pos);
        this.node.setPosition(this.pos.x, this.pos.y + this.direction * 80 * deltaTime, 0);
    }
}
```
This is how it looks when this demo is running in a browser (which is correct):

https://user-images.githubusercontent.com/82378750/219465509-6ea2f380-fbbf-460b-9b72-9efebaf834a4.mp4

But running the demo on Android devices the motionstreak does not appear and the screen is plain black.

This issue was not present in v3.5.2.

### Steps to reproduce

Run the attached demo project on Android to see the issue.

### Minimal reproduction project

[Demo2D.zip](https://github.com/cocos/cocos-engine/files/10759706/Demo2D.zip)

**Comment by mrtzd** [comment]
Ok so it seems like this issue has already been reported here (#12992) but for some reason was eventually ignored which is a bit surprising to me since this makes MotionStreak unusable.  

The bug was introduced with v3.6.0. Just from a cursory look it seems like that this pr (#11592) might have caused this regression.

**Comment by mrtzd** [comment]
Fixed by #14392
